ianfab-Fairy-Stockfish-c9bbf18 bug

@ddugovic @ianfab  
I'm opening here the issue according to @ianfab instruction https://github.com/ianfab/Fairy-Stockfish/issues/42#issuecomment-545482453

The windows worker stopped with this error:
```
Step 1/4. Building instrumented executable ...
make ARCH=x86-64-modern COMP=mingw gcc-profile-make
make[1]: Entering directory '/c/users/42/appdata/local/temp/tmp53jjtr/ianfab-Fairy-Stockfish-c9bbf18/src'
make ARCH=x86-64-modern COMP=mingw \
EXTRACXXFLAGS='-fprofile-generate' \
EXTRALDFLAGS='-lgcov' \
all
make[2]: Entering directory '/c/users/42/appdata/local/temp/tmp53jjtr/ianfab-Fairy-Stockfish-c9bbf18/src'
...
g++ -march=native -Wall -Wcast-qual -fno-exceptions -std=c++11  -Wextra -Wshadow -DANTI -DATOMIC -DBUGHOUSE -DCRAZYHOUSE -DDISPLACEDGRID -DEXTINCTION -DGIVEAWAY -DGRID -DHORDE -DKOTH -DLOOP -DLOSERS -DPLACEMENT -DRACE -DSLIPPEDGRID -DSUICIDE -DTHREECHECK -DTWOKINGS -DTWOKINGSSYMMETRIC -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -Wall -Wcast-qual -fno-exceptions -std=c++11  -Wextra -Wshadow -DANTI -DATOMIC -DBUGHOUSE -DCRAZYHOUSE -DDISPLACEDGRID -DEXTINCTION -DGIVEAWAY -DGRID -DHORDE -DKOTH -DLOOP -DLOSERS -DPLACEMENT -DRACE -DSLIPPEDGRID -DSUICIDE -DTHREECHECK -DTWOKINGS -DTWOKINGSSYMMETRIC -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -Wall -Wcast-qual -fno-exceptions -std=c++11 -fprofile-generate -Wextra -Wshadow -DANTI -DATOMIC -DBUGHOUSE -DCRAZYHOUSE -DDISPLACEDGRID -DEXTINCTION -DGIVEAWAY -DGRID -DHORDE -DKOTH -DLOOP -DLOSERS -DPLACEMENT -DRACE -DSLIPPEDGRID -DSUICIDE -DTHREECHECK -DTWOKINGS -DTWOKINGSSYMMETRIC -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT   -c -o syzygy/tbprobe.o syzygy/tbprobe.cpp
thread.cpp: In constructor 'Thread::Thread(size_t)':
thread.cpp:44:33: error: no matching function for call to 'NativeThread::NativeThread()'
   44 | Thread::Thread(size_t n) : idx(n) {
      |                                 ^
In file included from thread.h:39,
                 from thread.cpp:26:
thread_win32_osx.h:53:12: note: candidate: 'template<class T, class P> NativeThread::NativeThread(void (T::*)(), T*)'
   53 |   explicit NativeThread(void(T::*fun)(), T* obj) {
      |            ^~~~~~~~~~~~
thread_win32_osx.h:53:12: note:   template argument deduction/substitution failed:
thread.cpp:44:33: note:   candidate expects 2 arguments, 0 provided
   44 | Thread::Thread(size_t n) : idx(n) {
      |                                 ^
In file included from thread.h:39,
                 from thread.cpp:26:
thread_win32_osx.h:47:7: note: candidate: 'constexpr NativeThread::NativeThread(const NativeThread&)'
   47 | class NativeThread {
      |       ^~~~~~~~~~~~
thread_win32_osx.h:47:7: note:   candidate expects 1 argument, 0 provided
thread_win32_osx.h:47:7: note: candidate: 'constexpr NativeThread::NativeThread(NativeThread&&)'
thread_win32_osx.h:47:7: note:   candidate expects 1 argument, 0 provided
thread.cpp:47:51: error: no match for 'operator=' (operand types are 'NativeThread' and 'std::thread')
   47 |   stdThread = std::thread(&Thread::idle_loop, this);
      |                                                   ^
In file included from thread.h:39,
                 from thread.cpp:26:
thread_win32_osx.h:47:7: note: candidate: 'NativeThread& NativeThread::operator=(const NativeThread&)'
   47 | class NativeThread {
      |       ^~~~~~~~~~~~
thread_win32_osx.h:47:7: note:   no known conversion for argument 1 from 'std::thread' to 'const NativeThread&'
thread_win32_osx.h:47:7: note: candidate: 'NativeThread& NativeThread::operator=(NativeThread&&)'
thread_win32_osx.h:47:7: note:   no known conversion for argument 1 from 'std::thread' to 'NativeThread&&'
make[2]: *** [<builtin>: thread.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory
```

gcc version:
```
$ g++ -v
Using built-in specs.
COLLECT_GCC=C:\msys64\mingw64\bin\g++.exe
COLLECT_LTO_WRAPPER=C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../gcc-9.2.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,fortran,ada,objc,obj-c++ --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-filesystem-ts=yes --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --enable-plugin --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev2, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld
Thread model: posix
gcc version 9.2.0 (Rev2, Built by MSYS2 project)
```

--- a/src/thread.cpp
+++ b/src/thread.cpp
@@ -41,11 +41,13 @@ ThreadPool Threads; // Global object
 /// Thread constructor launches the thread and waits until it goes to sleep
 /// in idle_loop(). Note that 'searching' and 'exit' should be already set.
 
-Thread::Thread(size_t n) : idx(n) {
-
 #ifdef _WIN32
-  stdThread = std::thread(&Thread::idle_loop, this);
+Thread::Thread(size_t n) : idx(n), stdThread(&Thread::idle_loop, this) {
 #else
+Thread::Thread(size_t n) : idx(n) {
+#endif
+
+#ifndef _WIN32
   // With increased MAX_MOVES (for variants) the stack can grow larger than the
   // system default. Explicitly set a sufficient stack size.
   pthread_attr_t attr;

